bitcoin/src/blockdata/script.rs

```diff
@@ -400,6 +400,26 @@ impl Script {
       Builder::new()
     }
 
+    /// Returns the public key if this script is P2PK with a valid public key.
```

Comment:
Suggested addition (as second paragraph):

> **Important**: this may return `None` even when ``[`is_p2pk()`](Self::is_p2pk)`` returns `true`.
> This happens when the public key is invalid (e.g. the point not being on the curve).
> It also implies the script is unspendable.

Also I suggest putting emphasis on "valid" in summary, probably bold. (`**valid**`)

Reply:
Sure, clarifying like this makes sense